libs/framework/include/celix/dm/Component.h

```diff
@@ -52,6 +49,22 @@ namespace celix { namespace dm {
          * Returns the C bundle context
          */
         celix_bundle_context_t* bundleContext() const { return this->context; }
+
+        void runBuild();
+    protected:
+        std::vector<std::shared_ptr<BaseServiceDependency>> dependencies{};
+
+        // 0 = service name
```

Comment:
Perhaps worth considering just making a simple struct with names for the variables, rather than tuple. Then these comments are not needed anymore :)

Reply:
I created a ProvidedService class, now this is also more inline with ServiceDependency